src/main/java/practice/StreamPractice.java

```diff
@@ -23,7 +35,10 @@ public int findMinEvenNumber(List<String> numbers) {
      * But before that subtract 1 from each element on an odd position (having the odd index).
      */
     public Double getOddNumsAverage(List<Integer> numbers) {
-        return 0D;
+        return IntStream.range(0, numbers.size())
+                .map(i -> i % 2 == 0 ? numbers.get(i) : numbers.get(i) - 1)
+                .filter(f -> f % 2 != 0)
+                .average().orElseThrow(NoSuchElementException::new);
```

Comment:
it's optional to pass NoSuchElementException in the constructor of the method orElseThrow. It will be thrown by default
```suggestion
                .average()
                .orElseThrow();
```